sys/include/senml.h

```diff
@@ -0,0 +1,161 @@
+#define SJ_VALUE_SUM     "s"    //!< Key for the value sum attribute in JSON documents
+#define SJ_TIME          "t"    //!< Key for the time attribute in JSON documents
+#define SJ_UPDATE_TIME   "ut"   //!< Key for the update time attribute in JSON documents
+#define SJ_DATA_VALUE    "vd"   //!< Key for the data value attribute in JSON documents
+
+
+#define SC_VERSION       (-1)   //!< Key for the version attribute in CBOR documents
+#define SC_BASE_NAME     (-2)   //!< Key for the base name attribute in CBOR documents
+#define SC_BASE_TIME     (-3)   //!< Key for the base time attribute in CBOR documents
+#define SC_BASE_UNIT     (-4)   //!< Key for the base unit attribute in CBOR documents
+#define SC_BASE_VALUE    (-5)   //!< Key for the base value attribute in CBOR documents
+#define SC_NAME          (0)    //!< Key for the name attribute in CBOR documents
+#define SC_UNIT          (1)    //!< Key for the unit attribute in CBOR documents
+#define SC_VALUE         (2)    //!< Key for the value attribute in CBOR documents
+#define SC_STRING_VALUE  (3)    //!< Key for the string value attribute in CBOR documents
+#define SC_BOOL_VALUE    (4)    //!< Key for the boolean value attribute in CBOR documents
+#define SC_VALUE_SUM     (5)    //!< Key for the value sum attribute in CBOR documents
+#define SC_TIME          (6)    //!< Key for the time attribute in CBOR documents
+#define SC_UPDATE_TIME   (7)    //!< Key for the update time attribute in CBOR documents
+#define SC_DATA_VALUE    (8)    //!< Key for the data value attribute in CBOR documents
+
+
+/*! The different data types SenML supports */
+typedef enum {
+    SENML_TYPE_UNDEF  = 0,  //!< Placeholder for when no value is provided
+    SENML_TYPE_FLOAT  = 1,  //!< Indicates a float value
+    SENML_TYPE_STRING = 2,  //!< Indicates a string value
+    SENML_TYPE_BOOL   = 3,  //!< Indicates a boolean value
+    SENML_TYPE_BINARY = 4   //!< Indicates a binary (data) value
+} senml_value_type_t;
+
+
+/*! struct that contains base information which applies to all subsequent entries */
+typedef struct {
+    uint8_t             version;         //!< SenML version of this pack
+    char               *base_name;       //!< Concatenated with name attribute gives sensor's UID
+    double              base_time;       //!< Update time values are relative to this time
+    char               *base_unit;       //!< When all values have the same unit, set this field
+    double              base_value;
+} senml_base_info_t;
+
+
+/*! struct that contains the values of a SenML record */
+typedef struct {
+    char              *name;        //!< Sensor's name, will be appended to base name if provided
+    char              *unit;        //!< Unit of the measurement
+    double             time;        //!< Time (seconds since unix epoch) of the measurement
+    unsigned int       update_time; //!< Time before sensor provides an updated measurement
```

Comment:
Update time should be double (see [Section 8](https://tools.ietf.org/html/draft-ietf-core-senml-00#section-8)).